drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -414,14 +414,25 @@ static int nrf5_tx(const struct device *dev,
 
 	LOG_DBG("Result: %d", nrf5_data.tx_result);
 
-	if (nrf5_radio->tx_result == NRF_802154_TX_ERROR_NONE) {
+	switch (nrf5_radio->tx_result) {
+	case NRF_802154_TX_ERROR_NONE:
 		if (nrf5_radio->ack_frame.psdu == NULL) {
 			/* No ACK was requested. */
 			return 0;
 		}
-
 		/* Handle ACK packet. */
 		return handle_ack(nrf5_radio);
+	case NRF_802154_TX_ERROR_NO_MEM:
+		return -ENOMEM;
```

Comment:
Upper layers (like TCP) expects to receive `-ENOBUFS` in the case of out-of-buffers case. Could you change the `-ENOMEM` to `-ENOBUFS` here.

Reply:
done